client/dashboard/profile-wizard/steps/business-details.js

```diff
@@ -18,46 +18,41 @@ import { numberFormat } from '@woocommerce/number';
 /**
  * Internal dependencies
  */
-import { H, Card, SimpleSelectControl } from '@woocommerce/components';
+import { H, Card, SimpleSelectControl, Form } from '@woocommerce/components';
 import withSelect from 'wc-api/with-select';
 import { recordEvent } from 'lib/tracks';
 
 class BusinessDetails extends Component {
 	constructor() {
 		super();
 
-		this.state = {
-			errors: {},
-			fields: {
-				other_platform: '',
-				product_count: '',
-				selling_venues: '',
-				extensions: {
-					facebook: true,
-					mailchimp: true,
-				},
-			},
+		this.initialValues = {
+			other_platform: '',
+			product_count: '',
+			selling_venues: '',
+			facebook: true,
+			mailchimp: true,
 		};
 
+		this.extensions = [ 'facebook', 'mailchimp' ];
+
 		this.onContinue = this.onContinue.bind( this );
+		this.validate = this.validate.bind( this );
 	}
 
-	async onContinue() {
-		await this.validateForm();
-		if ( Object.keys( this.state.errors ).length ) {
-			return;
-		}
-
+	async onContinue( values ) {
 		const { createNotice, goToNextStep, isError, updateProfileItems } = this.props;
-		const { extensions, other_platform, product_count, selling_venues } = this.state.fields;
-		const businessExtensions = keys( pickBy( extensions ) );
+		const { facebook, mailchimp, other_platform, product_count, selling_venues } = values;
+		const businessExtensions = keys( pickBy( values ) ).filter( name =>
```

Comment:
Minor: Since this is done here, and line 110, should we pull it out into its own function?

Reply:
Good idea- added in https://github.com/woocommerce/woocommerce-admin/pull/2749/commits/56ce63fd770caf7d8bef9d2bd3a432e456632e21

Reply:
Just realized that was your commit that looks identical to mine. 🤓